postgis_eklentisi/postgis_eklentisi.md

```diff
@@ -0,0 +1,615 @@
+
+```sql
+-- Örnek Kullanım:
+
+SELECT ST_GeomFromKML('
+		<LineString>
+			<coordinates>-71.1663,42.2614
+				-71.1667,42.2616</coordinates>
+		</LineString>');
+```
+
+ST_AsKML fonksiyonu ile postgis geometrisinden OGC KML verisi elde edilebilir. Yine yalnızca geometri kısmı için geçerlidir. 
+
+## 7. TOPOLOJİK İLİŞKİLER
+---
+
+PostGIS, geometrilerin uzayda birbirine göre durumlarını sınayan bir dizi fonksiyona sahiptir. Bu fonksiyonlar kullanılarak mekansal veriler arasındaki ilişkileri sorgulayabilir aynı zamanda çeşitli mekansal analizleri gerçekleştirebiliriz.
+
+Bu bölümde geometrilerin birbirlerine göre durumlarını sınayan PostGIS fonksiyonları üzerinde durulmuştur. A ve B değişkenlerinin birer PostGIS geometrisi olduğu düşünülmelidir.
+
+### 7.1. ST_Contains(A, B)
+
+Girdi olarak iki adet geometri alan bu fonksiyon geometrilerin biribiri içinde olup olmadığını kontrol ederek sonucu boolean bir değer olarak döndürür. B geometrisi, tamamen A geometrisinin içerisinde ise True değer dönecektir [9]. Boolean değer döndürmesi sebebiyle where cümleciği ile ya da join işlemlerinde kullanılabilir.
+
+```sql
+-- Örnek Kullanım: Mahallelerdeki metro istasyonu sayılarını veren mekansal sorguda st_contains foksiyonu kullanılabilir.
+
+SELECT
+  neighborhoods.name AS neighborhood_name,
+  COUNT(*) AS number_of_subway
+FROM nyc_neighborhoods AS neighborhoods
+JOIN nyc_subway_stations AS subways
+ON ST_Contains(neighborhoods.geom, subways.geom)
+GROUP BY neighborhoods.name
+```
+
+### 7.2. ST_Crosses(A, B)
+
+Girdi olarak iki geometri alan bu fonksiyon ile geometrilerin uzayda kesişimi sınanır. Geometrilerin en az bir ortak noktaları varsa ve tümü ortak değilse true değer döndürür [9].
+
+```sql
+-- Örnek Kullanım: Mahalle sınırları ile kesişen yolların adı ve tipini getiren bir sorgu için st_crosses fonksiyonu kullanılabilir.
+
+SELECT streets.name, streets.type
+FROM nyc_streets as streets, nyc_neighborhoods as neighborhoods
+WHERE st_crosses(streets.geom, neighborhoods.geom)
+```
+
+### 7.3. ST_Disjoint(A, B)
+
+İki geometri için uzayda ayrıklığı sınayan fonksiyondur. Eğer paramere olarak alınan iki fonksiyon için bindirme ya da kesişme söz konusu değilse geometriler ayrıktır ve true değer döner. Aksi takdirde false değer dönecektir [9].
+
+```sql
+-- Örnek Kullanım:
+
+SELECT ST_Disjoint('POINT(0 0)'::geometry, 'LINESTRING ( 2 0, 0 2 )'::geometry) as result;
+```
+
+### 7.4. ST_Equal(A, B)
+
+İki geometri mekansal olarak eşit ise true değeri döndürür. Geometrilerin noktalarının sıralaması farklı olsa da aynı geometri yapısında olduğu için true döndürecektir. Eşit operatörüne göre daha iyi sonuç verir [9].
+
+```sql
+-- Örnek Kullanım:
+
+SELECT ST_Equals(ST_GeomFromText('LINESTRING(0 0, 10 10)'),
+    ST_GeomFromText('LINESTRING(0 0, 5 5, 10 10)')) as result;
+```
+
+### 7.5. ST_Intersects(A, B)
```

Comment:
st_intersects konum bazlı uygulamalarda sık kullanılır, çünkü çok performanslı çalışır. 
Çünkü gist index kullanan sorgulardan birisidir. Tam burada gist index ekleme ve kısace index ne işe yarar anlatmakta fayda var. Hatta anlatılmalı, Postgis'in en güçlü olduğu yer burası bence.
Bir de yine index kullanan st_dwithin sorgusunda bahsedilebilir.

Reply:
Gerekli eklemeleri yaptım.